### PR TITLE
Add support for Dependabot secrets

### PR DESCRIPTION
**Pull request by liath**
I added handling for Dependabot specific secrets per #2246. This part of the API seems undocumented, but I don't see why it would change as it's almost identical to managing regular secrets. This file is pretty repetitive, and I doubled the number of repo and org methods, so I did some refactoring. Hopefully nothing too heinous.

### CODE REVIEW

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:75]: Is there a better URL for this endpoint?

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:91]: Is there a better URL for this endpoint?

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:73]: Maybe a more specific description for this endpoint would be good?

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:89]: Maybe a more specific description for this endpoint would be good?

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:149]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:152]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:167]: and here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:170]: and here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:207]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:209]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:223]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:225]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:272]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:274]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:288]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:290]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:296]: Should this say "environment secret" instead of "repository secret"?
It looks like this may have been an pre-existing copy/paste mistake.

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:321]: This line needs updating.

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:323]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:337]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:339]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:387]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:389]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:416]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:418]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:441]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:443]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:466]: same here

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:468]: same here

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @liath !
Please add unit tests for the new methods.

File: github/actions_secrets.go
Comment (jurre) [reply github/actions_secrets.go:91]: https://docs.github.com/en/rest/reference/dependabot#get-an-organization-public-key

**Comment by jurre** [comment]
Slightly bike-sheddy, but I would suggest pulling this out of `github/actions_secrets.go` into `github/dependabot_secrets.go`. While the API shares a lot of similarities, it is it's own top-level resource that is geared towards Dependabot specifically, not Actions.

File: github/actions_secrets.go
Comment (liath) [reply github/actions_secrets.go:75]: I was going to say that this is an undocumented API, but docs were posted last night! https://github.com/dependabot/dependabot-core/issues/3297#issuecomment-1009308206

File: github/dependabot_secrets.go
Comment (gmlewis) [inline github/dependabot_secrets.go:1]:
```suggestion
// Copyright 2022 The go-github AUTHORS. All rights reserved.
```

Any time a new file is added, it is stamped with the current year it was created in.  Thanks.

File: github/dependabot_secrets_test.go
Comment (gmlewis) [inline github/dependabot_secrets_test.go:1]:
```suggestion
// Copyright 2022 The go-github AUTHORS. All rights reserved.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
This is looking great, @liath - thank you!
Just a couple minor tweaks and then I think we will be ready for a second LGTM/Approval before merging.

Note that any other contributor to this repo can review and give the second LGTM/Approval. Thank you!

**Review by gmlewis — CHANGES_REQUESTED** [review]
Oh, one more thing... since the new methods are part of the `ActionsService`, can you please rename the new files to `actions_dependabot_secrets.go` and `actions_dependabot_secrets_test.go`?  Thank you!

File: github/dependabot_secrets.go
Comment (liath) [reply github/dependabot_secrets.go:1]: Will do!

**Comment by liath** [comment]
Done! That feels much nicer, good suggestion :D

**Comment by liath** [comment]
Could we go a step further and make them `actions_secrets_dependabot*` so that all the secrets-management files are next to each other alphabetically?

**Comment by gmlewis** [comment]
> Could we go a step further and make them `actions_secrets_dependabot*` so that all the secrets-management files are next to each other alphabetically?

Genius!  Sorry I didn't think of that the first time, @liath !

**Comment by jurre** [comment]
Not super familiar with this project, but I do work on Dependabot (and implemented the API in question) and so I'm curious why this is part of the `ActionsService`, it seems more appropriate to add a new `DependabotService` struct, as I'd expect these APIs might diverge over time which could result in a breaking change to the library. Given that they are separate resources in the API, would it make sense for the client to follow that?

**Comment by gmlewis** [comment]
> Not super familiar with this project, but I do work on Dependabot (and implemented the API in question) and so I'm curious why this is part of the `ActionsService`, it seems more appropriate to add a new `DependabotService` struct, as I'd expect these APIs might diverge over time which could result in a breaking change to the library. Given that they are separate resources in the API, would it make sense for the client to follow that?
> 
> No strong opinion either way, confident y'all do what makes most sense for this project +1

Hi @jurre and thank you for the comments and questions (and for the API! :smile: )!

So in this repo, we try to tightly couple the layout and services to the GitHub v3 API documentation organization structure.

We initially placed these API endpoints under the `ActionsService` endpoints primarily because they appear to have a very high correlation with those endpoints, and because no `Dependabot` documentation was available at that time (if I'm not mistaken).

However, now that the API documentation exists and is indeed under a completely different section, I think you are right that we should create a new `DependabotService`.

@liath - would you like to tackle that, or should we address that in a follow-up PR (either by you or by another contributor)?

**Comment by jurre** [comment]
> very high correlation with those endpoints

You're right, that's definitely the case 👍 they share a lot of the same semantics, but might evolve separately

> no Dependabot documentation was available at that time

Yeah it took a few days to get those published so that's totally fair.

**Comment by liath** [comment]
Should I add Dependabot specific types like `DependabotSecrets` to fully separate them or would that create weirdness for callers?

**Comment by gmlewis** [comment]
I think we can keep the shared data structs for now (even though we are separating out the service)... with the understanding that if they diverge in the future, we can split it further.

**Comment by gmlewis** [comment]
And just to be clear, I think we can keep all that nice work you did to consolidate the shared code... again with the understanding that it may change some day, but I think it is super-clean for now, which is very nice.

**Review by gmlewis — APPROVED** [review]
Hey, this looks fantastic, @liath !  Thank you!
LGTM.
Awaiting second LGTM.

Maybe @Parker77 has time to review?

**Review by Parker77 — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**
